src/monumenten/_processing.py

```diff
@@ -0,0 +1,151 @@
+"""Interne processing functies voor de monumenten package."""
+
+import asyncio
+from typing import Any, Dict, List, Tuple
+
+import aiohttp
+import geopandas as gpd
+import pandas as pd
+from tqdm.asyncio import tqdm_asyncio
+
+from monumenten._api._cultureel_erfgoed import (
+    _query_beschermde_gebieden,
+    _query_rijksmonumenten,
+)
+from monumenten._api._kadaster import _query_verblijfsobjecten
+
+_QUERY_BATCH_GROOTTE = 500  # lijkt meest optimaal qua performance
+
+
+async def _verzamel_data(
+    session: aiohttp.ClientSession,
+    identificaties_batch: List[str],
+) -> Tuple[List[Dict[str, Any]], List[Dict[str, Any]]]:
+    """Verzamel data van verschillende API's voor een batch verblijfsobjecten.
+
+    Args:
+        session (aiohttp.ClientSession): De sessie voor HTTP requests
+        identificaties_batch (List[str]): Lijst met verblijfsobject ID's
+
+    Returns:
+        Tuple[List[Dict[str, Any]], List[Dict[str, Any]]]: Tuple met rijksmonumenten en verblijfsobjecten data
+    """
+    rijksmonumenten_taak = _query_rijksmonumenten(session, identificaties_batch)
+    verblijfsobjecten_taak = _query_verblijfsobjecten(session, identificaties_batch)
+    return await asyncio.gather(rijksmonumenten_taak, verblijfsobjecten_taak)
+
+
+async def _process_batch(
+    session: aiohttp.ClientSession, batch: List[str], bg_df: gpd.GeoDataFrame
+) -> Tuple[List[Dict[str, Any]], List[Dict[str, Any]], int]:
+    """Verwerk een batch verblijfsobjecten.
+
+    Args:
+        session (aiohttp.ClientSession): De sessie voor HTTP requests
+        batch (List[str]): Lijst met verblijfsobject ID's
+        bg_df (gpd.GeoDataFrame): GeoDataFrame met beschermde gezichten
+
+    Returns:
+        Tuple[List[Dict[str, Any]], List[Dict[str, Any]], int]: Tuple met rijksmonumenten,
+            beschermde gezichten en aantal verwerkte objecten
```

Comment:
Deze docstring is misschien niet heel verhelderend

Reply:
juiste typing in docstring is een vereiste :')

Reply:
Oh sorry, ik bedoelde: Tuple met rijksmonumenten, beschermde gezichten en aantal verwerkte objecten
Dat klinkt alsof er naast rijksmonumenten en beschermde gezichten ook "een aantal verwerkte objecten" teruggegeven wordt.

Reply:
hm ja, de functie returned `len(batch)` dus in principe returnt ie het aantal verwerkte objecten haha